lib/Sema/CSSimplify.cpp

```diff
@@ -3225,6 +3225,28 @@ bool ConstraintSystem::repairFailures(
     if (!isValueOfRawRepresentable(expectedType, rawReprType))
       return false;
 
+    auto rawValueLabel = getASTContext().Id_rawValue;
+    auto valueTy = expectedType->lookThroughSingleOptionalType();
+    auto unwrappedRawReprTy = rawReprType->lookThroughSingleOptionalType();
+    auto memberLoc = getConstraintLocator(
+        locator.withPathElement(ConstraintLocator::ApplyFunction));
+    auto memberTy = createTypeVariable(memberLoc, TVO_CanBindToNoEscape);
+    addValueMemberConstraint(
+        MetatypeType::get(unwrappedRawReprTy, getASTContext()),
+        DeclNameRef(DeclName(getASTContext(), DeclBaseName::createConstructor(),
+                             {rawValueLabel})),
+        memberTy, DC, FunctionRefKind::SingleApply, {},
+        getConstraintLocator(
+            locator.withPathElement(ConstraintLocator::ConstructorMember)));
+    addConstraint(
+        ConstraintKind::ApplicableFunction,
+        FunctionType::get({AnyFunctionType::Param(valueTy, rawValueLabel)},
+                          unwrappedRawReprTy),
+        memberTy, memberLoc);
+
+    ArgumentInfos[getArgumentInfoLocator(memberLoc)] =
+        ArgumentInfo({rawValueLabel, None});
```

Comment:
I'm not sure adding constraints here is correct – `matchTypes` will set up a disjunction for all the possible conversions and fixes, so we shouldn't really be adding these constraints until the `ExplicitlyConstructRawRepresentable` fix is attempted in `simplifyFixConstraint`.

Reply:
Yeah, this would be moved to `simplifyFixConstraint`, I just thought that @theblixguy is trying to get the logic right first...

Reply:
Yeah, I have moved it to `simplifyFixConstraint` now :)